Tests/LibBookmarkTests.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  LibBookmarkTests.swift
+//  UnitTests
+
+import XCTest
+@testable import Kiwix
+
+final class LibBookmarkTests: XCTestCase {
+
+    private let testZimURL: URL = Bundle(for: LibBookmarkTests.self).url(forResource: "test_small", withExtension: "zim")!
+    private let testTitle: String = "Dionysius of Halicarnassus On Literary Composition"
+    private let bookmarkURL = URL(string: "kiwix://FF7D59DE-0FD8-F486-09FA-C57904646707/A/Dionysius%20of%20Halicarnassus%20On%20Literary%20Composition.50212.html")!
+
+    func testIfZimFileExists() throws {
+        XCTAssertNotNil(testZimURL)
+    }
+
+    func testAddingBookmark() {
+        let fileURLData = ZimFileService.getFileURLBookmarkData(for: testZimURL)!
+        try! ZimFileService.shared.open(fileURLBookmark: fileURLData)
```

Comment:
This is a bit cumbersome, we have a hash of stored `Archive` under `ZimFileService` that came from parsing things, using `kiwix::library`` mostly (in `OPDSParser`), but we are not holding on to the `library`.
In order for the bookmarks to be successfully added, we do need to add them to the `library` itself.
So we have a kind of dualism here, where the list of items are going through one temporary instance of `library` to be parsed and inserted into the DB, whereas in order to store them we need another instance of `library` where the `kiwix:Book` had to be added, otherwise the added `kiwix::Bookmark` will be `invalid`.